app/forms/publish/course_location_form.rb

```diff
@@ -38,5 +35,22 @@ def no_locations_selected
 
       errors.add(:site_ids, :no_locations)
     end
+
+    def update_site_statuses
+      model.site_statuses.each do |site_status|
+        site_status.update!(
```

Comment:
Intentionally invoking ActiveRecord callbacks here to handle any additional side effects caused from saving